kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/embedder/MavenEmbedder.java

```diff
@@ -292,18 +300,29 @@ public String getLocation() {
         try {
-            Thread.currentThread().setContextClassLoader( componentProvider.getSystemClassLoader() );
-            ProjectBuilder projectBuilder = componentProvider.lookup( ProjectBuilder.class );
-            // BZ-1007894: Check if added dependencies are resolvable.
-            ProjectBuildingResult result = projectBuilder.build( modelSource, getProjectBuildingRequest() );
-            if ( result != null && result.getDependencyResolutionResult() != null && !result.getDependencyResolutionResult().getCollectionErrors().isEmpty() ) {
-                // A dependency resolution error has been produced. It can contains some error. Throw the first one to the client, so the user will fix every one sequentially.
-                Exception depedencyResolutionException = result.getDependencyResolutionResult().getCollectionErrors().get( 0 );
-                if ( depedencyResolutionException != null ) {
-                    throw new MavenEmbedderException( depedencyResolutionException.getMessage(), depedencyResolutionException );
+            org.eclipse.aether.artifact.Artifact lastArtifact = null;
+            do {
+                Thread.currentThread().setContextClassLoader(componentProvider.getSystemClassLoader());
+                ProjectBuilder projectBuilder = componentProvider.lookup(ProjectBuilder.class);
+                // BZ-1007894: Check if added dependencies are resolvable.
+                ProjectBuildingResult result = projectBuilder.build(modelSource, getProjectBuildingRequest());
+                if (result != null && result.getDependencyResolutionResult() != null && !result.getDependencyResolutionResult().getCollectionErrors().isEmpty()) {
+                    // A dependency resolution error has been produced. It can contains some error. Throw the first one to the client, so the user will fix every one sequentially.
+                    final Exception depedencyResolutionException = result.getDependencyResolutionResult().getCollectionErrors().get(0);
+                    if (depedencyResolutionException instanceof ArtifactDescriptorException) {
+                        final org.eclipse.aether.artifact.Artifact artifact = ((ArtifactDescriptorException) depedencyResolutionException).getResult().getArtifact();
+                        if (!artifact.equals(lastArtifact)) {
+                            tryRemoveLocalArtifact(artifact);
+                            lastArtifact = artifact;
+                            continue;
+                        }
+                    }
+                    if (depedencyResolutionException != null) {
+                        throw new MavenEmbedderException(depedencyResolutionException.getMessage(), depedencyResolutionException);
+                    }
                 }
-            }
-            return result.getProject();
+                return (result == null || result.getProject() == null ) ? null : result.getProject();
```

Comment:
After staring at the code for some time I didn't find much to improve.
But the code could be simplified by removing some unnecessary nullchecks (up to you).
According to javadoc ProjectBuilder#build
`@return The result of the project building, never {@code null}.`
so the `result!=null` at 309 and `result == null` at 234 can be removed.

Reply:
@jhrcek i don't have the confidence that ProjectBuilder#build won't ever be null.. would be better if it was wrapped in an Optional<T>, but would potentially be a bigger refactoring.. so I'd prefer to play safe here for now.